prusti-viper/src/encoder/foldunfold/footprint.rs

```diff
@@ -115,9 +115,6 @@ impl ExprFootprintGetter for vir::Expr {
                 box body,
                 ..
             }) => {
-                assert!(variables
```

Comment:
Is this meant to be part of this PR? If we do this we should look into actually using the `is_quantifiable` function (there is a TODO somewhere in the code mentioning that).

Reply:
When I removed the same assert that you mentioned in https://github.com/viperproject/prusti-dev/issues/56#issuecomment-1046966018 I ended up running into this one causing a panic. Since it was the same one I had already removed, I tried removing it as well. @vakaras I agree that maybe we don't want to entirely remove the asserts, but just modify `!var.typ.is_typed_ref_or_type_var()` to `!var.typ.is_typed_ref()` or something.

Reply:
> @vakaras I agree that maybe we don't want to entirely remove the asserts, but just modify `!var.typ.is_typed_ref_or_type_var()` to `!var.typ.is_typed_ref()` or something?

Yes.

Reply:
Removed by [this recent commit](https://github.com/viperproject/prusti-dev/commit/07f7d6ce5fb96faadbbb58122ae5052778f44b3c#diff-464a6cfdb5cefcf2b9e5ae3d8b06fadd22f506153b31b3791230611df9671b8aL328-L330) so this is no longer an issue